application/models/user.go

```diff
@@ -495,3 +496,27 @@ func (u *User) WantsPostNotification(post Post) bool {
 
 	return true
 }
+
+// GetPreferences returns a slice of matching UserPreference records
+func (u *User) GetPreferences() (UserPreferences, error) {
+	uPrefs := UserPreferences{}
+	err := DB.Where("user_id = ?", u.ID).All(&uPrefs)
+
+	return uPrefs, err
+}
+
+// GetPreference returns a pointer to a matching UserPreference record or if
+// none is found, returns nil
+func (u *User) GetPreference(key string) (*UserPreference, error) {
+	uPref := UserPreference{}
+
+	err := DB.Where("user_id = ?", u.ID).Where("key = ?", key).First(&uPref)
+	if err != nil {
+		if domain.IsErrorMoreThanJustNoSQLRows(err) {
```

Comment:
Since the function returns `false` for `err == nil`, do you need both `if`s?

Reply:
I think so, since if the error is not nil and not `NoRows` we return nil, err,
  if the error is not nil but is 'NoRows we return nil, nil, 
  and if the error is not nil, we return the object and nil.

Do you see a way of getting those three different things done with only one `if`?

Reply:
Would this work?
```go
if domain.IsErrorMoreThanJustNoSQLRows(err) {
    return nil, err
}
return &uPref, nil
```
Since the return value is probably nil when there's a no-rows error, or any error for that matter, this might be the same as what you have.

Reply:
Oh, I think I see what I missed before. You want to return a nil `*UserPreference` rather than a pointer to an empty `UserPreference` when there were no rows.